Improve documentation of harness files & raw flag (#3324)
tc39/test262: CONTRIBUTING.md

#### includes
`includes: [file-list]`

This key names a list of files in the `harness/` directory that will be included in the test environment prior to running the test.  Filenames **must** include the `.js` extension.

> When some code is used repeatedly across a group of tests, it may be appropriate to define it in a harness file. This practice increase test complexity, so it should be applied sparingly.

Reviewer: s/increase/increases/

- **noStrict** - only run the test in "sloppy" mode
- **module** - interpret the source text as
  [module code](https://tc39.github.io/ecma262/#sec-modules)

> - **raw** - execute the test without any modification (no harness files will be
>   included); necessary to test the behavior of directive prologue; implies

Reviewer 2: ```suggestion
- **raw** - execute the test without any modification (no harness files or extra content must be
  included); necessary to test the behavior of directive prologue; implies
```
This is a nit pick, but reflects better the correct functionality of the raw flagged files. If they will be included or not, that depends on the runner, which is not controlled by this repo.

## Test Environment

Each test case is run in a fresh JavaScript environment; in a browser, this will be a new &lt;iframe&gt;; for a console runner, this will be a new process.

> Before the test is executed (and unless the test uses the `raw` frontmatter flag), the test runner will evaluate a number of files in the `harness/` directory. At a minimum, this procedure will define the following functions:

Reviewer 2: The "at a minimum" part is slightly ambiguous. It's the minimum when raw flag is not present. I tried to rephrase the same content above to resolve this, WDYT?

```suggestion
The test runner must evaluate the required files from the `harness/` directory before the test is executed. Unless the `raw` frontmatter flag is present, this procedure will define at least the following functions:
```

var var = var;
```

> If the test case has the `raw` flag, the test runner will not load any harness file including the file which defines `$DONOTEVALUATE`. In this case, include a direct `throw "Test262: This statement should not be evaluated.";` statement:

Reviewer 2: ```suggestion
If the test case has the `raw` flag, the test runner must not load any harness file including the file which defines `$DONOTEVALUATE`. In this case, include a direct `throw "Test262: This statement should not be evaluated.";` statement:
```
